### ISSUE TITLE
x/build: plan9 unexpected EOF on listing files

### ISSUE THREAD

**Issue by bradfitz** [golang/go#10705]
Just noticed this:

```
$ gomote push plan9-386-gcepartial
Error running push: error listing buildlet's existing files: unexpected EOF
```

What's up? Investigate.

/cc @crawshaw @0intro @adg

**Comment by 0intro** [comment]
In the console, I can see:

```
Walk error: stat /tmp/buildlet-scatch854652968/src/image/testdata/video-001.separate.dc.progression.progre
```

The message is truncated, but it looks like the "stat buffer too short"
issue that was fixed in CL 3281.

I can reproduce the issue with http://storage.googleapis.com/go-builder-data/buildlet.plan9-386,
but not with a buildlet I built myself.

Are you sure your Plan 9 buildlet is up-to-date?

**Comment by bradfitz** [comment]
No, it's probably old.

Let me try again. I'm doing a demo in one hour. Hope I don't break it too much before the demo.

**Comment by 0intro** [comment]
Also, be sure to build the Plan 9 buildlet with Go 1.5.

**Comment by 0intro** [comment]
Good luck with your demo.

**Comment by bradfitz** [comment]
Okay, that was it. We should do a better job monitoring all buildlet versions.

Maybe the coordinator should show the latest of each it sees. /cc @crawshaw
